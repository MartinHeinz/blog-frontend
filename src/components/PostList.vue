<template>
    <div>
        <h1>Blog Posts</h1>
        <span>
            <ul v-if="postEmpty" class="post-list">
                <PostListItem v-for="item in postList"
                              v-bind:url="'/blog/'+ item.id"
                              v-bind:value="item.title"
                              v-bind:published="item.posted_on"
                              v-bind:key="item.id">
                </PostListItem>
            </ul>
            <p v-else>Coming soon...</p>
        </span>
    </div>
</template>

<script>
import PostListItem from '@/components/PostListItem.vue';
import { mapGetters, mapActions } from 'vuex';

export default {
    name: 'PostList',
    components: { PostListItem },
    props: {
        items: Array,
    },
    created() {
        this.fetchPosts();
    },
    computed: {
        ...mapGetters([
            'postList',
        ]),
        postEmpty() {
            return this.postList.length > 0;
        },
    },
    methods: {
        ...mapActions([
            'fetchPosts',
        ]),
    },
};
</script>

<style scoped>
    @media (min-width: 480px) {
        .post-list .post-item {
            display: flex;
            margin-bottom: 5px;
        }
        .post-list .post-item .meta {
            text-align: left;
        }
    }
</style>
