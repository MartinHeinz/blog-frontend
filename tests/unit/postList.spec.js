import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils';
import PostList from '@/components/PostList.vue';
import moment from 'moment';

const items = [
    {
        id: 1,
        created_at: '0001-01-01T00:00:00Z',
        updated_at: '0001-01-01T00:00:00Z',
        deleted_at: null,
        title: 'First Blog Post',
        text: 'This is blog about something.',
        author: 'Martin',
        next: null,
        next_post_id: 2,
        previous: null,
        previous_post_id: 0,
        posted_on: '0001-01-01T00:00:00Z',
        sections: null,
        tags: null,
    },
    {
        id: 3,
        created_at: '0001-01-01T00:00:00Z',
        updated_at: '0001-01-01T00:00:00Z',
        deleted_at: null,
        title: '3rd Blog Post',
        text: 'Another dummy content',
        author: 'Martin',
        next: null,
        next_post_id: 0,
        previous: null,
        previous_post_id: 2,
        posted_on: '0001-01-01T00:00:00Z',
        sections: null,
        tags: null,
    },
];

describe('PostList.vue', () => {
    it('renders correcly', () => {
        const localVue = createLocalVue();
        localVue.filter('formatDate', value => moment(String(value)).format('ll')); // TODO import this from @/main.js
        localVue.component('router-link', RouterLinkStub);
        const wrapper = mount(PostList, {
            localVue,
            propsData: {
                items,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
