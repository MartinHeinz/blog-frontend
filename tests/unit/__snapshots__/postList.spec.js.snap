// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList.vue renders correcly 1`] = `
<div>
  <h1>Blog Posts</h1> <span><ul class="post-list"><li class="post-item"><div class="meta"><time datetime="0001-01-01T00:00:00Z" itemprop="datePublished">Jan 1, 0001</time></div> <span><a>First Blog Post</a></span></li>
  <li class="post-item">
    <div class="meta"><time datetime="0001-01-01T00:00:00Z" itemprop="datePublished">Jan 1, 0001</time></div> <span><a>3rd Blog Post</a></span>
  </li>
  </ul>
  <!---->
  </span>
</div>
`;
